### PR TITLE
[v6r15] Several bug fixes

### PR DESCRIPTION
**Pull request by atsareg**
- mysql startup script is in the new location with the new MySQL version
- restored RequiredTag logic in the Matcher

### CODE REVIEW

File: FrameworkSystem/Client/ComponentInstaller.py [hunk 1]
```diff
@@ -184,6 +184,12 @@ def loadDiracCfg( self, verbose = False ):
     self.mysqlMyCnf = os.path.join( self.mysqlDir, '.my.cnf' )
 
     self.mysqlStartupScript = os.path.join( rootPath, 'mysql', 'share', 'mysql', 'mysql.server' )
+    # With the new versions of MySQL server the startup script is in a different location
+    if not os.path.exists( self.mysqlStartupScript ):
+      startupScript = os.path.join( rootPath, 'mysql', 'support-files', 'mysql.server' )
+      if os.path.exists( startupScript ):
+        os.makedirs( os.path.dirname( self.mysqlStartupScript ) )
+        shutil.copy( startupScript, self.mysqlStartupScript )
```
Comment (fstagni) [inline FrameworkSystem/Client/ComponentInstaller.py]: I think a link may be more elegant than a copy.

Comment (chaen) [inline FrameworkSystem/Client/ComponentInstaller.py]: agreed

File: Resources/Computing/SSHComputingElement.py [hunk 2]
```diff
@@ -271,6 +271,8 @@ def _reset( self ):
     """ Process CE parameters and make necessary adjustments
     """
     self.batchSystem = self.ceParameters.get( 'BatchSystem', 'Host' )
+    if not 'BatchSystem' in self.ceParameters:
```
Comment (chaen) [inline Resources/Computing/SSHComputingElement.py]: Cosmetic, but if I am not mistaken, pep 8 prefers "not in"

File: FrameworkSystem/Client/ComponentInstaller.py (hunk 1 above)
Comment (atsareg) [inline FrameworkSystem/Client/ComponentInstaller.py]: Actually, I found the way to place the startup scripts in the same location while compiling Externals. So, I will revert back this change

**Pull request merged**